compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -29,6 +31,8 @@ use self::debuginfo::{FunctionDebugContext, PerLocalVarDebugInfo};
 use self::operand::{OperandRef, OperandValue};
 use self::place::PlaceRef;
 
+const MIN_DANGEROUS_SIZE: u64 = 1024 * 1024 * 1024 * 1; // 1 GB
```

Comment:
Not to change, but to discuss: would making this lower than 1GB be worth it, so that people that we can guide people that are not hitting the limits, but are on their way to do so get some early warning?

Reply:
What I remember from before (about 9 months back) that this is around the limit of where the overflow in some calculations of memory allocations go wrong in LLVM. There were a few tests pointing to the same, in the original issue description. It could very well not be true now, but I think the higher level issue when it's resolved in LLVM and we no longer need this warning, is the way to go.